pkg/batch/jobs/calculator/experiment_lock.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 The Bucketeer Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package calculator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/bucketeer-io/bucketeer/pkg/lock"
+	redisv3 "github.com/bucketeer-io/bucketeer/pkg/redis/v3"
+)
+
+const (
+	experimentLockPrefix = "experiment-calculate:"
+	experimentLockTTL    = time.Minute
```

Comment:
Depending on the number of running experiments, it could take more than 1 minute to finish analyzing the data.
It will be safer if we set the default value to 10 minutes.
Also, can you make this configurable? It will be nice to pass the config via values.yaml.